pkg/controller/identity/iamaccesskey/controller.go

```diff
@@ -72,7 +72,15 @@ type connector struct {
 }
 
 func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
-	cfg, err := awsclient.GetConfig(ctx, c.kube, mg, awsclient.GlobalRegion)
+	cr, ok := mg.(*v1alpha1.IAMAccessKey)
+	region := awsclient.GlobalRegion
+	if cr.Spec.ForProvider.Region != "" {
+		region = cr.Spec.ForProvider.Region
+	}
+	if !ok {
```

Comment:
This check needs to happen right after `cr, ok := mg.(*v1alpha1.IAMAccessKey)` otherwise cr is `nil`